rust-workspace/crates/ntc-oracle-node/src/signer.rs

```diff
@@ -0,0 +1,38 @@
+use ed25519::signature::{Signer, Verifier};
+
+pub(crate) struct AuthDataSigner<S>
+where
+    S: Signer<ed25519::Signature>,
+{
+    pub signing_key: S,
+}
+
+impl<S> AuthDataSigner<S>
+where
+    S: Signer<ed25519::Signature>,
+{
+    pub(crate) fn sign(&self, auth_data: &[u8]) -> ed25519::Signature {
+        // NOTE: use `try_sign` if you'd like to be able to handle
+        // errors from external signing services/devices (e.g. HSM/KMS)
+        // <https://docs.rs/signature/latest/signature/trait.Signer.html#tymethod.try_sign>
+        self.signing_key.sign(auth_data)
+    }
+}
+#[allow(dead_code)]
+pub(crate) struct AuthDataVerifier<V> {
+    pub verify_key: V,
+}
+
+impl<V> AuthDataVerifier<V>
+where
+    V: Verifier<ed25519::Signature>,
+{
+    #[allow(dead_code)]
+    pub(crate) fn verify(
+        &self,
+        auth_data: &[u8],
+        signature: &ed25519::Signature,
+    ) -> Result<(), ed25519::Error> {
+        self.verify_key.verify(auth_data, signature)
+    }
+}
```

Comment:
Does this code not get executed?

Reply:
I see, that makes sense.  If it only gets used in the unit tests then we might rather want to use the `#[cfg(test)]` attribute.  That way it only gets compiled for the tests and is not included in the release artifacts.